pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Stage.groovy

```diff
@@ -106,11 +109,23 @@ public class Stage implements NestedModel, Serializable {
      * @return a list of "key=value" strings.
      */
     List<String> getEnvVars() {
-        return environment.collect { k, v ->
+        return environment.findAll{k, v -> !(v instanceof CredentialWrapper)}.collect { k, v ->
             "${k}=${v}"
         }
     }
 
+    @Whitelisted
+    @Nonnull
+    Map<String, CredentialWrapper> getEnvCredentials() {
+        Map<String, CredentialWrapper> m = [:]
+        environment.each {k, v ->
```

Comment:
I don't know, it seems to be since it works as well as the code in the method above that has been working for a long time.